src/projects/EnsureThat/ExceptionFactory.cs

```diff
@@ -27,8 +29,8 @@ public static Exception CreateForComparableParamValidation<T>(Param<T> param, st
                     : string.Concat(message, Environment.NewLine, param.ExtraMessageFn(param)));
         }
 
-        [Pure]
```

Comment:
Not pure, as it can throw. Not changing the throw to be a return b/c it would lose stack trace that may be intended in the ExceptionFn